classes/battle/BattleAttack.php

```diff
@@ -6,6 +6,15 @@
 
 class BattleAttack {
     public Jutsu $jutsu;
-    public float $raw_damage;
-    public float $countered_percent = 0;
+    public float $raw_damage = 0;
+    public float $piercing_percent = 0;
+    public float $substitution_percent = 0;
+    public float $counter_percent = 0;
+    public float $immolate_percent = 0;
+    public float $immolate_raw_damage = 0;
+    public float $recoil_percent = 0;
+    public float $recoil_raw_damage = 0;
+    public float $countered_percent = 0; // opponent's counter
+    public float $countered_raw_damage = 0; // damage dealt from opponent
+    public string $countered_jutsu_type; // jutsu type of opponent's counter
```

Comment:
This is great, nice to keep everything organized with the attack object itself 🎉